server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -82,13 +81,14 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.function.LongSupplier;
 
 final class DefaultSearchContext extends SearchContext {
 
     private final long id;
     private final ShardSearchRequest request;
     private final SearchShardTarget shardTarget;
-    private final Counter timeEstimateCounter;
+    private final LongSupplier timeEstimate;
```

Comment:
I think this name is misleading because we are using a relative notion of time here whereas the name could be construed as an estimate of the absolute time (like a timestamp). In other places we use the name `relativeTimeSupplier` to make it clear the purpose of the variable, and to avoid a problem if in the future someone were to use this variable for another piece of code to try to get the absolute time. It would also be worthwhile to add a comment explaining why relative time is okay here. It is important to keep the two uses between absolute and relative time absolutely clear.

Reply:
Should we change also the iface method name(): https://github.com/elastic/elasticsearch/pull/39048/files#diff-2410736cbe252badfd50ccbfceebc857R398 ?

Reply:
I would just name it timeSupplier. I also think we should not expose the LongSupplier on SerachContext at all and only provide a method to access the time `public long getTime()` on `SearchContext.java` The usage is simple and can just call this method. no need to expose the supplier.

Reply:
Chose `getTimeInMillis()` to be even more clear.

Reply:
Please include the word `relative` in the name.